sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -942,6 +942,34 @@ class FileSourceCharVarcharTestSuite extends CharVarcharTestSuite with SharedSpa
       }
     }
   }
+
+  test("SPARK-48498: always do char padding in predicates") {
+    import testImplicits._
+    withSQLConf(SQLConf.READ_SIDE_CHAR_PADDING.key -> "false") {
```

Comment:
If set key `LEGACY_NO_CHAR_PADDING_IN_PREDICATE` = true, this case will fail.

if not set this key to true, maybe can not get data when char column compare to liter, because liter would be padded but  char column not,  any idea?